fetch.bs

```diff
@@ -7234,22 +7246,28 @@ of running <a for=Body>consume body</a> with <a>this</a> and the following step
 <p>The <dfn method for=Body><code>blob()</code></dfn> method steps are to return the result
 of running <a for=Body>consume body</a> with <a>this</a> and the following step given a
 <a for=/>byte sequence</a> <var>bytes</var>: return a {{Blob}} whose contents are <var>bytes</var>
-and whose {{Blob/type}} attribute is <a>this</a>'s <a for=Body>MIME type</a>.
+and whose {{Blob/type}} attribute is to <a for=Body>get the MIME type</a> with <a>this</a>.
 </div>
 
 <div algorithm>
 <p>The <dfn method for=Body><code>formData()</code></dfn> method steps are to return the result of
 running <a for=Body>consume body</a> with <a>this</a> and the following step given a
-<a for=/>byte sequence</a> <var>bytes</var>: switch on <a>this</a>'s <a for=Body>MIME type</a>'s
-<a for="MIME type">essence</a> and run the corresponding steps:
+<a for=/>byte sequence</a> <var>bytes</var>:
 
-<dl class=switch>
+<ol>
+<li><p>Let <var>mimeType</var> be the result of <a for=Body>get the MIME type</a> with <a>this</a>.
+<li><p>If <var>mimeType</var> is null, then <a>Throw</a> a {{TypeError}}.
+<li>
+  <p>Otherwise, switch on <var>mimeType</var>'s <a for="MIME type">essence</a> and run the
+  corresponding steps:
```

Comment:
Here's what I think I would do

| If mimeType is non-null, then switch on _mimeType_'s essence and run the ...

and then you remove the "Otherwise" from the list and instead end with a "Throw" step.

The one unfortunate thing about all this is that you'll have to adjust indentation, which might require rewrapping due to 100 columns. If that seems like too much of a challenge I'm happy to do that for you though near the end.